Source/Lib/Trakt.NET/Extensions/IAsyncEnumerableExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace TraktNet.Extensions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>Provides helper methods for IAsyncEnumerables.</summary>
+    public static class IAsyncEnumerableExtensions
+    {
+        /// <summary>Converts the given IAsyncEnumberable to a list and returns that list.</summary>
+        /// <param name="source">The IAsyncEnumerable to convert to a list</param>
+        /// <param name="cancellationToken">
+        /// Propagates notification that the request should be canceled.<para/>
+        /// If provided, the exception <see cref="OperationCanceledException" /> should be catched.
+        /// </param>
+        /// <returns>A list contains the values ion the IAsyncEnumerable</returns>
+        /// <exception cref="ArgumentException">Thrown, if the given string is null or empty.</exception>
+        public static async Task<List<TSource>> ToListAsync<TSource>(this IAsyncEnumerable<TSource>  source, CancellationToken cancellationToken = default)
+        {
+            if (source is null)
+                return default;
```

Comment:
Shouldn't here be returned an empty list already, even if source is null?
Even if source is null, the caller can still proceed (with an empty list) but does not need to take care of a possible null reference exception.

Reply:
It does return a List<TSource>. It's returning the default of the return type. It's the same as the TraktClient taking in a default version of the IHttpClientProvider.

Reply:
HttpClientProvider = httpClientProvider ?? new HttpClientProvider(this);

Reply:
But `List<TSource>` is a reference type, so as defined in the [.NET documentation](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/default-values) for default values
```csharp
List<TSource> result = default;
```
would mean that `result` is `null` and therefore in this case `return default` would mean `return null`, or am I missing something?

You are correct with `IHttpClientProvider httpClientProvider = default` in the `TraktClient` constructor.
But there is also this line:
```csharp
HttpClientProvider = httpClientProvider ?? new HttpClientProvider(this);
```
which checks if the given `httpClientProvider` is `null` and creates a new instance if it is `null`.

Reply:
I don't know how I blanked on that. You are correct.

Reply:
Happens to the best of us ☺️